atc/worker/placement_test.go

```diff
@@ -652,62 +669,62 @@ var _ = Describe("LimitActiveVolumesPlacementStrategyNode", func() {
 					)
 					Expect(chooseErr).ToNot(HaveOccurred())
 					return chosenWorker
-				}).Should(Or(Equal(compatibleWorker1), Equal(compatibleWorker2), Equal(compatibleWorker3)))
+				}).Should(Or(Equal(compatibleWorker1), Equal(compatibleWorker2)))
 			})
 		})
 
-		Context("when there is a limit", func() {
-			Context("when the limit is 20", func() {
-				BeforeEach(func() {
-					activeVolumeLimit = 20
-				})
+		Context("when the memory is above the max limit", func() {
+			BeforeEach(func() {
+				worker1AllocatedMemory = 1500
+				worker2AllocatedMemory = 0
+				requestedMemory = 1000
+			})
 
-				It("picks worker1", func() {
-					Consistently(func() Worker {
-						chosenWorker, chooseErr = strategy.Choose(
-							logger,
-							workers,
-							spec,
-						)
-						Expect(chooseErr).ToNot(HaveOccurred())
-						return chosenWorker
-					}).Should(Equal(compatibleWorker1))
-				})
+			It("return only workers with enough available memory", func() {
+				Consistently(func() Worker {
+					chosenWorker, chooseErr = strategy.Choose(
+						logger,
+						workers,
+						spec,
+					)
+					Expect(chooseErr).ToNot(HaveOccurred())
+					return chosenWorker
+				}).Should(Equal(compatibleWorker2))
+			})
+		})
 
-				Context("when the limit is 200", func() {
-					BeforeEach(func() {
-						activeVolumeLimit = 200
-					})
+		Context("when a worker does not configure allocatable memory", func() {
+			var noAllocatableMemoryWorker *workerfakes.FakeWorker
 
-					It("picks worker1 or worker2", func() {
-						Consistently(func() Worker {
-							chosenWorker, chooseErr = strategy.Choose(
-								logger,
-								workers,
-								spec,
-							)
-							Expect(chooseErr).ToNot(HaveOccurred())
-							return chosenWorker
-						}).Should(Or(Equal(compatibleWorker1), Equal(compatibleWorker2)))
-					})
-				})
+			BeforeEach(func() {
+				worker1AllocatedMemory = 1500
+				worker2AllocatedMemory = 1500
+				requestedMemory = 1000
```

Comment:
wondering what if `requestedMemory` is below 500 in this case i.e. when there is worker with configured memory limit satisfying and also worker without configured allocatable memory available, which worker will be chosen?